### PR TITLE
Make "user_namespace" workspaces default

### LINKED ISSUE #3375: Make "user_namespace" workspaces default
We want to roll out user-namespaced workspaces generally. We should remove the feature flag and remove code pathes for the alternatives.

### PR DESCRIPTION
**Pull request by aledbf**
fixes #3375

How to test: running `sudo docker-up` in any example without enabling feature preview must work.

### CODE REVIEW

File: components/ws-manager/pkg/manager/create.go [hunk 1]
```diff
@@ -336,6 +371,26 @@ func (m *Manager) createDefiniteWorkspacePod(startContext *startWorkspaceContext
...
+			},
+			corev1.VolumeMount{
+				MountPath:        "/.workspace",
+				Name:             daemonVolumeName,
+				MountPropagation: &mountPropagation,
+			},
+		)
+		pod.Spec.Containers[i].Command = []string{"/.supervisor/workspacekit", "ring0"}
+		break
+	}
```
Comment (csweichel) [inline components/ws-manager/pkg/manager/create.go:392]: This should be folded into `createWorkspaceContainer`

File: components/ws-manager/pkg/manager/testdata/cdwp_duplicate_featureflag.golden [hunk 2]
```diff
@@ -1,236 +1,12 @@
...
-                "gitpod/servicePrefix": "foobarservice",
-                "gitpod/traceid": "",
-                "gitpod/url": "test-foobarservice-gitpod.io",
-                "gitpod/withUsernamespace": "true",
-                "prometheus.io/path": "/metrics",
-                "prometheus.io/port": "23000",
-                "prometheus.io/scrape": "true",
-                "seccomp.security.alpha.kubernetes.io/pod": "localhost/workspace-default"
-            }
+            "creationTimestamp": null
```
Comment (csweichel) [inline components/ws-manager/pkg/manager/testdata/cdwp_duplicate_featureflag.golden:4]: This looks like a broken test to me. Looks like the "double feature flag" we were using was "user_namespace".

File: components/ws-manager/pkg/manager/create.go (hunk 1 above)
Comment (aledbf) [reply components/ws-manager/pkg/manager/create.go]: fixed

File: components/ws-manager/pkg/manager/testdata/cdwp_duplicate_featureflag.golden (hunk 2 above)
Comment (aledbf) [reply components/ws-manager/pkg/manager/testdata/cdwp_duplicate_featureflag.golden]: fixed

File: components/ws-daemon/pkg/content/service.go [hunk 3]
```diff
@@ -475,7 +471,7 @@ func (s *WorkspaceService) uploadWorkspaceContent(ctx context.Context, sess *ses
 		}()
 
 		var opts []archive.TarOption
-		if sess.UserNamespaced && !sess.FullWorkspaceBackup {
+		if !sess.FullWorkspaceBackup {
```
Comment (csweichel) [inline components/ws-daemon/pkg/content/service.go:474]: When we bring this into production, we need to think about the non-user-namespaced workspaces that are currently running and will be backed up by this new version of ws-daemon. We should maintain the different behaviour in ws-daemon until after the next deployment (first week task to clean things up).

**Review by csweichel — COMMENTED** [review]
Other than the post-deployment migration (i.e. handling of non-user-namespaced workspaces right after the deployment), things look good.

**Review by csweichel — APPROVED** [review]
Sorry for the long turn-around times.

Change LGTM.

**Pull request merged**